File: gin-admin-backend/internal/service/analysis.go
package service

import (
	"gin-admin-backend/internal/data"
	"gin-admin-backend/pkg/utils"
	"net/http"

	"github.com/gin-gonic/gin"
)

type AnalysisController struct{}

func NewAnalysisController() *AnalysisController {
	return &AnalysisController{}
}

// GetTotal returns total counts
func (c *AnalysisController) GetTotal(ctx *gin.Context) {
	var userCount int64
	data.DB.Model(&data.User{}).Count(&userCount)

	// Mock data for other stats
	ctx.JSON(http.StatusOK, utils.Result{
		Code: 0,
		Msg:  "success",
		Data: gin.H{
			"user":    userCount,
			"views":   10240,
			"message": 321,
			"money":   8888,
		},
	})
}

// GetUserAccessSource returns user access source data
func (c *AnalysisController) GetUserAccessSource(ctx *gin.Context) {
	ctx.JSON(http.StatusOK, utils.Result{
		Code: 0,
		Msg:  "success",
		Data: []gin.H{
			{"value": 1048, "name": "Search Engine"},
			{"value": 735, "name": "Direct"},
			{"value": 580, "name": "Email Marketing"},
			{"value": 484, "name": "Union Ads"},
			{"value": 300, "name": "Video Ads"},
		},
	})
}

// GetWeeklyUserActivity returns weekly user activity
func (c *AnalysisController) GetWeeklyUserActivity(ctx *gin.Context) {
	ctx.JSON(http.StatusOK, utils.Result{
		Code: 0,
		Msg:  "success",
		Data: []gin.H{
			{"value": 13253, "name": "Mon"},
			{"value": 34235, "name": "Tue"},
			{"value": 26321, "name": "Wed"},
			{"value": 12340, "name": "Thu"},
			{"value": 24643, "name": "Fri"},
			{"value": 1322, "name": "Sat"},
			{"value": 1324, "name": "Sun"},
		},
	})
}

// GetMonthlySales returns monthly sales data
func (c *AnalysisController) GetMonthlySales(ctx *gin.Context) {
	ctx.JSON(http.StatusOK, utils.Result{
		Code: 0,
		Msg:  "success",
		Data: []gin.H{
			{"estimate": 100, "actual": 120, "name": "Jan"},
			{"estimate": 120, "actual": 82, "name": "Feb"},
			{"estimate": 161, "actual": 91, "name": "Mar"},
			{"estimate": 134, "actual": 154, "name": "Apr"},
			{"estimate": 105, "actual": 162, "name": "May"},
			{"estimate": 160, "actual": 140, "name": "Jun"},
			{"estimate": 165, "actual": 145, "name": "Jul"},
			{"estimate": 114, "actual": 250, "name": "Aug"},
			{"estimate": 163, "actual": 134, "name": "Sep"},
			{"estimate": 185, "actual": 56, "name": "Oct"},
			{"estimate": 118, "actual": 99, "name": "Nov"},
			{"estimate": 123, "actual": 123, "name": "Dec"},
		},
	})
}

File: python-admin-backend/app/api/analysis.py
from fastapi import APIRouter
from app.schemas.analysis import AnalysisTotal, UserAccessSource, WeeklyUserActivity, MonthlySales
from app.schemas.common import ResponseModel

router = APIRouter()


@router.get("/total", response_model=ResponseModel)
async def get_analysis_total():
    return ResponseModel(
        code=0,
        data=AnalysisTotal(users=1234, messages=5678, moneys=99999, shoppings=8888),
        message="success",
    )


@router.get("/userAccessSource", response_model=ResponseModel)
async def get_user_access_source():
    return ResponseModel(
        code=0,
        data=[
            UserAccessSource(value=3350, name="Direct"),
            UserAccessSource(value=3100, name="Email"),
            UserAccessSource(value=2340, name="Ad Networks"),
            UserAccessSource(value=1350, name="Video Ads"),
            UserAccessSource(value=2480, name="Search Engines"),
        ],
        message="success",
    )


@router.get("/weeklyUserActivity", response_model=ResponseModel)
async def get_weekly_user_activity():
    return ResponseModel(
        code=0,
        data=[
            WeeklyUserActivity(value=150, name="Mon"),
            WeeklyUserActivity(value=230, name="Tue"),
            WeeklyUserActivity(value=224, name="Wed"),
            WeeklyUserActivity(value=218, name="Thu"),
            WeeklyUserActivity(value=135, name="Fri"),
            WeeklyUserActivity(value=147, name="Sat"),
            WeeklyUserActivity(value=260, name="Sun"),
        ],
        message="success",
    )


@router.get("/monthlySales", response_model=ResponseModel)
async def get_monthly_sales():
    return ResponseModel(
        code=0,
        data=[
            MonthlySales(name="Jan", estimate=1000, actual=1200),
            MonthlySales(name="Feb", estimate=1100, actual=950),
            MonthlySales(name="Mar", estimate=1300, actual=1450),
            MonthlySales(name="Apr", estimate=1200, actual=1100),
            MonthlySales(name="May", estimate=1400, actual=1350),
            MonthlySales(name="Jun", estimate=1500, actual=1600),
        ],
        message="success",
    )
